src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorParser.cs

```diff
@@ -87,29 +87,7 @@ public class MvcRazorParser : RazorParser
                 _viewImportsDirectiveDescriptors,
                 errorSink);
 
-            var descriptors = visitor.GetDescriptors(documentRoot);
-            foreach (var descriptor in descriptors)
-            {
-                foreach (var attributeDescriptor in descriptor.Attributes)
-                {
-                    if (attributeDescriptor.IsIndexer &&
-                        string.Equals(
-                            attributeDescriptor.TypeName,
-                            _modelExpressionTypeName,
-                            StringComparison.Ordinal))
-                    {
-                        errorSink.OnError(
-                            SourceLocation.Undefined,
-                            Resources.FormatMvcRazorParser_InvalidPropertyType(
-                                descriptor.TypeName,
-                                attributeDescriptor.Name,
-                                _modelExpressionTypeName),
```

Comment:
We avoid optional parameters in almost every case except `private` helpers. See [https://github.com/aspnet/Home/wiki/Engineering-guidelines#async-method-patterns]() for another corner case exception. And, a change is not mandated here -- just nice to have. Leave the signature of the `MvcRazorHost` constructor as it was.